src/components/file-input/file-input.component.tsx

```diff
@@ -0,0 +1,256 @@
+      setIsDraggingFile(false);
+    };
+
+    useEffect(() => {
+      document.addEventListener("dragover", startDrag);
+      document.addEventListener("drop", stopDrag);
+
+      return () => {
+        document.removeEventListener("dragover", startDrag);
+        document.removeEventListener("drop", stopDrag);
+      };
+    }, []);
+
+    const onSelectFileClick = () => {
+      internalInputRef.current?.click();
+    };
+
+    const onFileAdded = (files: FileList) => {
+      onChange?.(files);
+    };
+
+    const onDragOver = (e: React.DragEvent<HTMLDivElement>) => {
+      e.preventDefault();
+      if (!disabled) {
+        if (e.dataTransfer?.types.includes("Files")) {
+          setIsDraggedOver(true);
+        }
+      }
+    };
+
+    const onDragLeave = (e: React.DragEvent<HTMLDivElement>) => {
+      e.preventDefault();
+      setIsDraggedOver(false);
+    };
+
+    const onDrop = (e: React.DragEvent<HTMLDivElement>) => {
+      e.preventDefault();
+      if (!disabled) {
+        setIsDraggedOver(false);
+        onFileAdded(e.dataTransfer.files);
+      }
+    };
+
+    const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+      onFileAdded(e.target.files as FileList);
+    };
+
+    const { labelId, validationId } = useInputAccessibility({
+      id: uniqueId,
+      validationRedesignOptIn: true,
+      error,
+      label,
+    });
+
+    // allow for single input that a single set of status props is provided
+    const filesUploaded = Array.isArray(uploadStatus)
+      ? uploadStatus
+      : [uploadStatus];
+
+    const input = (
+      <>
+        {inputHint && <StyledHintText>{inputHint}</StyledHintText>}
+        <Box position="relative">
+          <ValidationMessage error={error} validationId={validationId} />
+          {error && <ErrorBorder />}
+          <StyledHiddenFileInput
+            {...(required && { required })}
+            accept={accept}
+            aria-invalid={!!error}
+            id={uniqueId}
+            ref={internalCallbackRef}
+            name={uniqueName}
+            onChange={onInputChange}
+            type="file"
+            {...rest}
+          />
+          <StyledFileInputPresentation
+            isDisabled={disabled}
+            isDraggedOver={isDraggedOver}
+            isDraggingFile={isDraggingFile}
+            error={error}
+            onDragLeave={onDragLeave}
+            onDragOver={onDragOver}
+            onDrop={onDrop}
+            {...sizeProps}
+          >
+            <Typography m={0}>{mainText}</Typography>
+            <ButtonMinor
+              buttonType="primary"
+              onClick={onSelectFileClick}
+              disabled={disabled}
+            >
+              {textOnButton}
+            </ButtonMinor>
+          </StyledFileInputPresentation>
+        </Box>
+      </>
+    );
+
+    return (
+      // do not support old-style validation for File Input component
+      <CarbonProvider validationRedesignOptIn>
```

Comment:
good question - I added this at an early stage after seeing that things like InputPresentation checked the flag. But I didn't end up using that component - so it may be that this isn't necessary any more. I'll check - it would be nice to get rid of it as it will hopefully be going away soon(ish) anyway!